src/shared/middlewares/phoenix-receiver-middleware.js

```diff
@@ -0,0 +1,97 @@
+import { MESSAGE } from 'phoenix-middleware';
+import { protocol } from 'message-factory';
+import { startCountdown, startRound, finishRound } from '../actions/round-phase-actions';
+import { setTimeRemaining } from '../actions/countdown-actions';
+import { receivePuzzle, receivePuzzlesCount } from '../actions/puzzle-actions';
+import { receiveUser } from '../actions/user-actions';
+import { puzzle } from '../selectors/puzzle-selectors';
+
+const { MESSAGE_NAME } = protocol.ui;
+
+const roundPhaseToActionMap = {
+    'countdown': startCountdown,
+    'in-progress': startRound,
+    'end': finishRound,
+};
```

Comment:
We also have an `idle` round phase

Reply:
UI doesn't have separate view or way to handle this phase, so why bother 😺